pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExportCoordinator.java

```diff
@@ -0,0 +1,326 @@
+
+        withAPIsTxn(handle -> {
+            ExportUtil.computeMaxInstancesSeen(handle, activityExtracts);
+            ExportUtil.computeActivityAttributesSeen(handle, activityExtracts);
+            return null;
+        });
+        boolean runSuccess = runCsvExports(customDto, activityExtracts);
+
+        if (!runSuccess) {
+            ApiResult res = client.sendMail(createNotificationEmail(true, false));
+            if (res.hasThrown()) {
+                LOG.error("Sending notification email for study " + customDto.getGuid() + " with failed custom export failed",
+                        res.getThrown());
+            }
+            return false;
+        }
+
+        ApiResult res = client.sendMail(createNotificationEmail(false, false));
+        if (res.hasThrown()) {
+            LOG.error("Error while sending custom export email notification", res.getThrown());
+            return false;
+        }
+
+        return true;
+    }
+
+    private Mail createNotificationEmail(boolean isFailure, boolean isSkip) {
+        Config cfg = exportCfg.getConfig(CustomExportConfigFile.EMAIL);
+        String subject;
+        Email fromEmail = new Email(cfg.getString(CustomExportConfigFile.EMAIL_FROM_EMAIL),
+                cfg.getString(CustomExportConfigFile.EMAIL_FROM_NAME));
+        Email toEmail = new Email(cfg.getString(CustomExportConfigFile.EMAIL_TO_EMAIL),
+                cfg.getString(CustomExportConfigFile.EMAIL_TO_NAME));
+
+        List<Content> content = new ArrayList<>();
+
+        if (isFailure) {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_ERROR_SUBJECT);
+            content.add(new Content("text/plain", cfg.getString(CustomExportConfigFile.EMAIL_ERROR_CONTENT)));
+        } else if (isSkip) {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_SKIP_SUBJECT);
+            content.add(new Content("text/plain", cfg.getString(CustomExportConfigFile.EMAIL_SKIP_CONTENT)));
+        } else {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_SUCCESS_SUBJECT);
+            String fileUrl =
+                    "https://console.cloud.google.com/storage/browser/" + exportCfg.getString(CustomExportConfigFile.BUCKET_NAME)
+                            + "/" + exportCfg.getString(CustomExportConfigFile.FILE_PATH);
+            addSuccessContent(content, cfg.getString(CustomExportConfigFile.EMAIL_SUCCESS_CONTENT), fileUrl);
+        }
+
+        return createMail(subject, fromEmail, toEmail, content);
+    }
+
+    private void addSuccessContent(List<Content> content, String fullContent, String fileUrl) {
+        Content linkContent = new Content("text/html", "<a href=\"" + fileUrl + "\">" + fileUrl + "</a>");
+        if (!fullContent.contains("{{bucketLink}}")) {
+            content.add(new Content("text/plain", fullContent));
+            content.add(linkContent);
+        } else {
+            int startLinkIndex = fullContent.indexOf("{{bucketLink}}");
```

Comment:
Is this how things are done in Gen2 version of this export? Can we perhaps pass in the whole content to SendGrid, pass in data as part of the `Personalization` object, and leverage SendGrid's dynamic templating?

Reply:
Good idea